### PR TITLE
Vendor new Vault API package

### PR DESCRIPTION
**Pull request by dadgar**
Fixes https://github.com/hashicorp/nomad/issues/4633

### CODE REVIEW

File: vendor/vendor.json [hunk 1]
```diff
@@ -196,11 +196,12 @@
 		{"path":"github.com/hashicorp/serf/coordinate","checksumSHA1":"0PeWsO2aI+2PgVYlYlDPKfzCLEQ=","revision":"80ab48778deee28e4ea2dc4ef1ebb2c5f4063996","revisionTime":"2018-05-07T23:19:28Z"},
 		{"path":"github.com/hashicorp/serf/serf","checksumSHA1":"QrT+nzyXsD/MmhTjjhcPdnALZ1I=","revision":"80ab48778deee28e4ea2dc4ef1ebb2c5f4063996","revisionTime":"2018-05-07T23:19:28Z"},
 		{"path":"github.com/hashicorp/vault","checksumSHA1":"eGzvBRMFD6ZB3A6uO750np7Om/E=","revision":"182ba68a9589d4cef95234134aaa498a686e3de3","revisionTime":"2016-08-21T23:40:57Z"},
```
Comment (schmichael) [inline vendor/vendor.json]: Should we keep the root pkg matching as well?

Comment (schmichael) [reply vendor/vendor.json]: Actually it's a main pkg... Probably shouldn't vendor it at all?

Comment (dadgar) [reply vendor/vendor.json]: Yeah I removed the top level from vendor

**Comment by dadgar** [comment]
@preetapan circonus was using an older version of our retryable http client and the vendored vault api was relying on the new one. So there was a build error unless I vendored circonus such that both were using the new API

**Pull request merged**